src/ClassicUO.Assets/MapLoader.cs

```diff
@@ -539,47 +373,230 @@ public unsafe bool ApplyPatches(ref StackDataReader reader)
+        {
+            Maps = new OneMapLoader[MAPS_COUNT];
+        }
 
-                if (_filesMap[i] != null && _filesMap[i] is not UOFileUop && _filesMap[i].HasData)
+        public unsafe Task Load()
+        {
+            return Task.Run
+            (
+                () =>
                 {
-                    if (_filesIdxStatics[i] != null && _filesIdxStatics[i].HasData)
+                    bool foundOneMap = false;
+
+                    if (!string.IsNullOrEmpty(MapsLayouts))
                     {
-                        if (_filesStatics[i] != null && _filesStatics[i].HasData)
+                        string[] values = MapsLayouts.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+
+                        MAPS_COUNT = values.Length;
+                        MapsDefaultSize = new int[values.Length, 2];
+
+                        Log.Trace($"default maps size overraided. [count: {MAPS_COUNT}]");
+
+
+                        int index = 0;
+
+                        char[] splitchar = new char[1] { ',' };
+
+                        foreach (string s in values)
                         {
-                            for (int block = 0; block < maxBlockCount; block++)
+                            string[] v = s.Split(splitchar, StringSplitOptions.RemoveEmptyEntries);
+
+                            if (v.Length >= 2 && int.TryParse(v[0], out int width) && int.TryParse(v[1], out int height))
                             {
-                                ref IndexMap index = ref list[block];
-                                index.MapAddress = index.OriginalMapAddress;
-                                index.StaticAddress = index.OriginalStaticAddress;
-                                index.StaticCount = index.OriginalStaticCount;
+                                MapsDefaultSize[index, 0] = width;
+                                MapsDefaultSize[index, 1] = height;
+
+                                Log.Trace($"overraided map size: {width},{height}  [index: {index}]");
                             }
+                            else
+                            {
+                                Log.Error($"Error parsing 'width,height' values: '{s}'");
+                            }
+
+                            ++index;
                         }
                     }
+
+
+                    Initialize();
+
+                    for (var i = 0; i < MAPS_COUNT; ++i)
+                    {
+                        Maps[i] = new OneMapLoader(i);
+
+                        if (Maps[i].Exists)
+                            foundOneMap = true;
+                    }
+
+                    if (!foundOneMap)
+                    {
+                        throw new FileNotFoundException("No maps found.");
+                    }
+
+
+                    int mapblocksize = sizeof(MapBlock);
+
+                    if (Maps[0].MapFileLength / mapblocksize == 393216 || UOFileManager.Version < ClientVersion.CV_4011D)
```

Comment:
We don't need to handle CV_4011D versions. Let's get rid of this.

Reply:
You mean support for all clients older than CV_4011D should be removed?
Okay for me, I can write a patch after merging this. In this PR. I havn't changed any features, just moving existing code around.

Reply:
Do you want to remove support for verdata as well? (<CV_500A)
Require CV_5020 or CV_5090? (There's no CV_5080)

Reply:
Yes let's remove less than 5.0.0a and verdata

Reply:
What about this expression - is the `||` a typo, or was it really meant that way? I don't know what this kludge really means. I wish there were more code comments about stuff like that.

Reply:
Yeah that `||` I believe is correct. It used to be that we determined if the map files are "old format" or "new format" by size or block count. It's a trash way of doing it.

Reply:
So ... can we remove the whole block and remove the size check because this should only be reachable with V4 data files?

Reply:
The map changed from 6144 to 7168 when ML came out. This coincides with 4.0.11d+. I think back then, people tried to use old map files with newer clients. So that is correct, we can scrap this check entirely!